ecommerce/models.py

```diff
@@ -74,10 +78,20 @@ class ProductVersion(TimestampedModel):
     )
     price = models.DecimalField(decimal_places=2, max_digits=20)
     description = models.TextField()
+    text_id = models.TextField(null=True)
 
     class Meta:
         indexes = [models.Index(fields=["created_on"])]
 
+    def save(self, *args, **kwargs):  # pylint: disable=arguments-differ
+        if hasattr(self.product.content_object, "readable_id"):
```

Comment:
I think we'd be better off with a property on the `Program` and `CourseRun` models (`product_text_id` or something like that. This could then be written as:

```python
try:
  self.text_id = getattr(self.product.content_object, "product_text_id")
except AttributeError:
  log.error("The content object for this ProductVersion (%s) does not have a `product_text_id` property", str(self.id))
super().save(*args, **kwargs)
```

One additional benefit: in the model definitions it's clearer which ones are used as products.

Reply:
Rather than add an additional field to each model which will/should have the same exact value as `courserun.courseware_id` or `program.readable_id`, seems like it would be better to rename both of those to `text_id.  But I'm not sure if there's a particular reason why the fields are named what they currently are, maybe for consistency with edx?  @pdpinch @ferdi what would you prefer to do here?

Reply:
Not an additional field, a property like `page` or `first_unexpired_run`. It would return `courseware_id` in `CourseRun`, and `readable_id` in `Program`

Reply:
Thanks, I get it now, will update